lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module

```diff
@@ -335,3 +376,26 @@ function dosomething_rogue_check_rbid_and_store_ref($rogue_reportback) {
 
    return $data;
  }
+
+/**
+ * Query to check if a reportback item exists in the dosomething_rogue_failed_task_log table.
+ *
+ * @param string $northstar_id
+ * @param int $campaign_run_id
+ * @param string $caption
+ * @param string $file
+ *
+ */
+function dosomething_rogue_rb_previously_failed($northstar_id, $campaign_run_id, $caption, $file) {
```

Comment:
I think you can just pass an array into this function and then when you call it, pass the values array in. Then you can grab the values you need to run the query from there.